File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpItemUseOnBlockData.java
package calebxzhou.rdi.mc.common2.mcp;

public record RMcpItemUseOnBlockData(
        String action,
        boolean dryRun,
        RMcpBlockPosData pos,
        String face,
        String requestedItemId,
        Integer requestedInventorySlot,
        String hand,
        int requestedTimes,
        int performedTimes,
        boolean changedBlock,
        String beforeBlockId,
        String afterBlockId,
        String beforeBlockState,
        String afterBlockState,
        RMcpInventoryData.Item itemBefore,
        RMcpInventoryData.Item itemAfter,
        RMcpInventoryData inventory
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpItemUseOnBlockData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpItemUseOnBlockData(
    val action: String,
    val dryRun: Boolean,
    val pos: RBlockPos,
    val face: String,
    val requestedItemId: String,
    val requestedInventorySlot: Int,
    val hand: String,
    val requestedTimes: Int,
    val performedTimes: Int,
    val changedBlock: Boolean,
    val beforeBlockId: String,
    val afterBlockId: String,
    val beforeBlockState: String,
    val afterBlockState: String,
    val itemBefore: RMcpInventoryData.Item,
    val itemAfter: RMcpInventoryData.Item,
    val inventory: RMcpInventoryData?
)
